selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/details/WorkspaceDetailsMachineActionsTest.java

```diff
@@ -0,0 +1,267 @@
+    // check machine name editing
+    workspaceDetailsMachines.clickOnEditButton(MACHINE_NAME);
+    editMachineForm.waitForm();
+    /*editMachineForm.typeName(CHANGED_MACHINE_NAME);
+    editMachineForm.waitNameValidHighlighting();
+    editMachineForm.waitSaveButtonEnabling();*/
+
+    // check RAM field behavior with min valid value
+    editMachineForm.typeRam(MIN_VALID_RAM_VALUE);
+    editMachineForm.waitSliderRamValue(MIN_VALID_RAM_VALUE + " GB");
+    seleniumWebDriverHelper.sendKeys(ARROW_DOWN.toString());
+    editMachineForm.waitSliderRamValue(MIN_VALID_RAM_VALUE + " GB");
+    editMachineForm.waitRamFieldText(MIN_VALID_RAM_VALUE);
+
+    seleniumWebDriverHelper.sendKeys(ARROW_UP.toString());
+    editMachineForm.waitRamFieldText("1");
+    editMachineForm.waitSliderRamValue("1 GB");
+
+    // check RAM behavior with more than max valid value
+    editMachineForm.typeRam(BIGGER_THAN_VALID_RAM_SIZE);
+    editMachineForm.waitSaveButtonDisabling();
+    editMachineForm.waitSliderRamValue("GB");
+
+    // check RAM behavior with max valid value
+    editMachineForm.typeRam("100");
+    editMachineForm.waitSaveButtonEnabling();
+    editMachineForm.waitSliderRamValue(MAX_VALID_RAM_VALUE + " GB");
+
+    seleniumWebDriverHelper.sendKeys(ARROW_UP.toString());
+    editMachineForm.waitNameValidHighlighting();
+    editMachineForm.waitSaveButtonEnabling();
+    editMachineForm.waitRamFieldText(MAX_VALID_RAM_VALUE);
+    editMachineForm.waitSliderRamValue(MAX_VALID_RAM_VALUE + " GB");
+
+    seleniumWebDriverHelper.sendKeys(ARROW_DOWN.toString());
+    editMachineForm.waitNameValidHighlighting();
+    editMachineForm.waitSaveButtonEnabling();
+    editMachineForm.waitRamFieldText("99.5");
+    editMachineForm.waitSliderRamValue("99.5 GB");
+
+    // check restoring of default values after clicking on "Cancel" button
+    editMachineForm.typeRam("4");
+    editMachineForm.waitSaveButtonEnabling();
+    editMachineForm.waitSliderRamValue("4 GB");
+    editMachineForm.clickOnCancelButton();
+    editMachineForm.waitFormInvisibility();
+    workspaceDetailsMachines.clickOnEditButton(MACHINE_NAME);
+    editMachineForm.waitForm();
+    waitRecipeText(IMAGE_NAME);
+
+    // check saving of the changes
+    editMachineForm.typeRam(CHANGED_RAM_SIZE);
+    editMachineForm.waitRamFieldText(CHANGED_RAM_SIZE);
+    editMachineForm.waitSaveButtonEnabling();
+    editMachineForm.clickOnSaveButton();
+    editMachineForm.waitFormInvisibility();
+    workspaceDetails.waitEnabled(SAVE_BUTTON, APPLY_BUTTON, CANCEL_BUTTON);
+    workspaceDetailsMachines.waitMachinesListItemWithAttributes(
+        MACHINE_NAME, IMAGE_NAME, CHANGED_RAM_SIZE);
+    workspaceDetails.waitEnabled(SAVE_BUTTON, APPLY_BUTTON, CANCEL_BUTTON);
+    workspaceDetails.waitAndClickOn(SAVE_BUTTON);
+    workspaceDetailsMachines.waitMachinesListItemWithAttributes(
+        MACHINE_NAME, IMAGE_NAME, CHANGED_RAM_SIZE);
+  }
+
+  @Test
+  public void checkSettingsButton() {
```

Comment:
Looks like the test is checking not just settings button. Therefore the test name `checkMachineSettings` would be  more suitable here.